code/go/internal/validator/semantic/validate_minimum_kibana_version.go

```diff
@@ -0,0 +1,153 @@
+		return ve.ValidationErrors{err}
+	}
+
+	packageVersion, err := getPackageVersion(*manifest)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	kibanaVersionCondition, err := getKibanaVersionCondition(*manifest)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	err = validateMinimumKibanaVersion(packageType, *packageVersion, kibanaVersionCondition)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	return nil
+}
+
+func validateMinimumKibanaVersion(packageType string, packageVersion semver.Version, kibanaVersionCondition string) error {
+
+	if packageType != "input" {
+		return nil
+	}
+
+	if packageVersion.LessThan(semver.MustParse("1.0.0")) {
+		return nil
+	}
+
+	if kibanaVersionConditionIsGreaterThanOrEqualTo8_8_0(kibanaVersionCondition) {
+		return nil
+	}
+
+	return errors.New("Warning: conditions.kibana.version must be ^8.8.0 or greater for non experimental input packages (version > 1.0.0)")
+}
+
+func readManifest(fsys fspath.FS) (*pkgpath.File, error) {
+	manifestPath := "manifest.yml"
+	f, err := pkgpath.Files(fsys, manifestPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't locate manifest file")
+	}
+
+	if len(f) != 1 {
+		return nil, errors.New("single manifest file expected")
+	}
+
+	return &f[0], nil
+}
+
+func getPackageType(manifest pkgpath.File) (string, error) {
+
+	val, err := manifest.Values("$.type")
+	if err != nil {
+		return "", errors.Wrap(err, "can't read manifest type")
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return "", errors.New("manifest type is not a string")
+	}
+
+	return sVal, nil
+}
+
+func getPackageVersion(manifest pkgpath.File) (*semver.Version, error) {
+
+	val, err := manifest.Values("$.version")
+	if err != nil {
+		return nil, errors.Wrap(err, "can't read manifest version")
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return nil, errors.New("package version is not a string")
+	}
+
+	sVersion, err := semver.NewVersion(sVal)
+	return sVersion, nil
+}
+
+func getKibanaVersionCondition(manifest pkgpath.File) (string, error) {
+
+	val, err := manifest.Values("$.conditions[\"kibana.version\"]")
+	if err != nil {
+		return "", nil
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return "", errors.New("manifest kibana version is not a string")
+	}
+
+	return sVal, nil
+}
+
+func kibanaVersionConditionIsGreaterThanOrEqualTo8_8_0(kibanaVersionCondition string) bool {
+	if kibanaVersionCondition == "" {
+		return false
+	}
+
+	if kibanaVersionCondition == "^8.8.0" {
+		return true
+	}
+
+	// get all versions e.g 8.8.0, 8.8.1 from "^8.8.0 || ^8.8.1" and check if any of them is less than 8.8.0
+	pattern := `(\d+\.\d+\.\d+)`
+	semver8_8_0 := semver.MustParse("8.8.0")
+	regex := regexp.MustCompile(pattern)
+	matches := regex.FindAllString(kibanaVersionCondition, -1)
+
+	for _, match := range matches {
+		matchVersion, err := semver.NewVersion(match)
+		if err != nil {
+			return false
+		}
+
+		if matchVersion.LessThan(semver8_8_0) {
+			return false
+		}
+	}
```

Comment:
Maybe this can be changed to use `semver.NewConstraint`? 

https://pkg.go.dev/github.com/Masterminds/semver/v3#hdr-Checking_Version_Constraints_and_Comparing_Versions

There is an example in elastic-package that just do those checks (there the kibana version is already in the struct):
- https://github.com/elastic/elastic-package/blob/main/internal/packages/conditions.go#L28

IIRC and following that example in elastic-package, the requirement kibana version would be "8.8.0" and the constraint the one defined in the manifest.

Reply:
Will do 👍

Reply:
Doing some quick tests about this, probably it could not be achieved using semver.NewConstraint. It would be like comparing two constraints, the one from the package (kibana.version) against `>= 8.8.0`, and I think that could not be done with that.